File: client/cpp/casinoClient.cpp
#include <iostream>
#include "connection.h"
#include "json.hpp"
using json = nlohmann::json;

std::string pwd;
std::string clientName = "cppCasinoClient";
int currentRound;
int totalSlot;
int winningSlot;
int switchLeft;
int currentWealth;
bool isStart =false;
bool casinoTurn=true;
bool gameOver=false;
bool changedSlot=false;

void getStatus(){
    std::string ret = sendGet("http://localhost:8080/casino/status?pwd="+ pwd);
    std::cout<<ret<<std::endl;
    json j = json::parse(ret);
    currentWealth = j["deposit"];
    totalSlot = j["totalSlot"];
    winningSlot = j["winningSlot"];
    switchLeft = j["switchLeft"];
    casinoTurn = j["casinoTurn"];
    isStart = j["start"];
    gameOver =j["gameOver"];
    currentRound = j["currentRound"];
    changedSlot = j["changedSlot"];
}

void sendMove(int newWinningSLot){
    std::string ret = sendPost("http://localhost:8080/casino/sendMove?pwd="+pwd+"&winningSlot="+ std::to_string(newWinningSLot));
    std::cout<<ret<<std::endl;
    json j = json::parse(ret);
    totalSlot = j["totalSlot"];
    winningSlot = j["winningSlot"];
    switchLeft = j["switchLeft"];
    casinoTurn = j["casinoTurn"];
    isStart = j["start"];
    gameOver =j["gameOver"];
    currentRound = j["currentRound"];
    changedSlot = j["changedSlot"];
}
int main() {
    initCurl();
    pwd=sendPost("http://localhost:8080/casino/join?name="+ clientName);
    getStatus();
    while(!isStart){
        getStatus();
    }
    
    while(!gameOver){
        if(casinoTurn){
            //TODO: change the strategy
            if(switchLeft) {
                sendMove((currentRound - 1) % 100 + 1);
            }
            else{
                sendMove(winningSlot);
            }
        }
        getStatus();
    }
    cleanCurl();
    return 0;
}

File: client/python/casinoClient.py
import json
import requests
URL = "http://localhost:8080"
name = "pythonCasinoClient"


def sendPost(path):
    r = requests.post(URL+path)
    return r.text


def sendGet(path):
    r = requests.get(URL+path)
    return r.text


def getStatus(pwd):
    ret = sendGet("/casino/status?pwd="+pwd)
    return json.loads(ret)


def sendMove(pwd, winningSlot):
    ret = sendPost("/casino/sendMove?pwd="+pwd+"&winningSlot="+str(winningSlot))
    return json.loads(ret)


if __name__ == '__main__':
    pwd = sendPost("/casino/join?name="+name)

    status = getStatus(pwd)
    while not status["start"]:
        status = getStatus(pwd)

    #TODO: set the firstWiningSlot
    status = sendMove(pwd, 1)
    while not status["gameOver"]:
        if status["casinoTurn"]:
            #TODO: set winnigSlot
            sendMove(pwd, 1)
        status = getStatus(pwd)


